tartiflette/schema/schema.py

```diff
@@ -582,8 +631,217 @@ def _validate_directive_implementation(self) -> List[str]:
+                            f"cause Interface already exists."
+                        )
+
+                errors.extend(
+                    _validate_extension_directives(extension, extended, "TYPE")
+                )
+
+        return errors
+
+    def _validate_union_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLUnionTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLUnionType, "UNION"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                for typ in extension.types:
+                    if typ in extended.types:
+                        errors.append(
+                            f"Can't add PossibleType < {typ} > to "
+                            f"UNION < {extended.name} > "
+                            f"cause PossibleType already exists."
+                        )
+
+                errors.extend(
+                    _validate_extension_directives(
+                        extension, extended, "UNION"
+                    )
+                )
+
+        return errors
+
+    def _validate_input_object_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLInputObjectTypeExtension)
+        ]:
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLInputObjectType, "INPUT"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                errors.extend(
+                    _validate_extension_directives(
+                        extension, extended, "INPUT"
+                    )
+                )
+
+                for ifield in extension.input_fields:
+                    if ifield in extended.input_fields:
+                        errors.append(
+                            f"Can't add Input Field < {ifield} > "
+                            f"to Input Object < {extended.name} > "
+                            f"cause it already exists"
+                        )
+
+        return errors
+
+    def _validate_interface_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLInterfaceTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLInterfaceType, "INTERFACE"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                for field in extension.fields:
```

Comment:
Using `already_defined` function, this could be rewrite as:
```python
errors.extend(
    already_defined(
        extended.implemented_fields,
        extension.fields,
        extended.name,
        "Can't add field < {existing_value} > to interface < {extended_name} > cause field already exists.",
    )
)
```